### PR TITLE
Simplifies/cleans up constructors of GCE caching agents

### PR DESCRIPTION
**Pull request by ttomsu**
@duftler @lwander PTAL.

tag: https://github.com/spinnaker/spinnaker/issues/712

### CODE REVIEW

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/AbstractGoogleCachingAgent.groovy [hunk 1]
```diff
@@ -22,11 +22,11 @@ import com.google.api.client.googleapis.batch.BatchRequest
 import com.google.api.client.http.HttpRequest
 import com.google.api.client.http.HttpRequestInitializer
 import com.google.api.services.compute.Compute
+import com.netflix.servo.util.VisibleForTesting
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/AbstractGoogleCachingAgent.groovy]: I think this is supposed to be `com.google.common.annotations.VisibleForTesting`.

**Comment by duftler** [comment]
LGTM after addressing comment.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/AbstractGoogleCachingAgent.groovy [hunk 2]
```diff
@@ -37,13 +37,34 @@ abstract class AbstractGoogleCachingAgent implements CachingAgent, AccountAware
...
   GoogleCloudProvider googleCloudProvider
   String googleApplicationName // "Spinnaker/${version}" HTTP header string
-  String accountName
   GoogleNamedAccountCredentials credentials
-  // TODO(ttomsu): Make project and compute dynamic getters from credentials object.
-  String project
-  Compute compute
   ObjectMapper objectMapper
 
+  @VisibleForTesting
```
Comment (cfieber) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/AbstractGoogleCachingAgent.groovy]: you probably want an `@PackageScope` on here too if you really only want this accessible for a test case (its public by default in the wild wild groovy west)

Comment (ttomsu) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/AbstractGoogleCachingAgent.groovy]: Meant to pull in the google commons one.

**Pull request merged**